src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -289,6 +169,24 @@ private SocketAsyncEngine()
             }
         }
 
+        private void AddToMap(SocketAsyncContext context)
+        {
+            IntPtr socketFileDescriptor = context.GetSocketFileDescriptor();
```

Comment:
This is using `DangerousGetHandle` which may cause issues (besides 5e64fa3) due to fd recycling.